packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -140,51 +140,48 @@ module.exports = (
           planId,
         } = await request.app.metricsContext;
 
-        return customs
-          .check(request, email, 'accountCreate')
-          .then(deleteAccountIfUnverified)
-          .then(setMetricsFlowCompleteSignal)
-          .then(generateRandomValues)
-          .then(createPassword)
-          .then(createAccount)
-          .then(createSessionToken)
-          .then(sendVerifyCode)
-          .then(createKeyFetchToken)
-          .then(recordSecurityEvent)
-          .then(createResponse);
-
-        function deleteAccountIfUnverified() {
-          return db
-            .getSecondaryEmail(email)
-            .then(secondaryEmailRecord => {
-              // Currently, users can not create an account from a verified
-              // secondary email address
-              if (secondaryEmailRecord.isPrimary) {
-                if (secondaryEmailRecord.isVerified) {
-                  throw error.accountExists(secondaryEmailRecord.email);
-                }
-                request.app.accountRecreated = true;
-                return db.deleteAccount(secondaryEmailRecord).then(() =>
-                  log.info('accountDeleted.unverifiedSecondaryEmail', {
-                    ...secondaryEmailRecord,
-                  })
-                );
-              } else {
-                if (secondaryEmailRecord.isVerified) {
-                  throw error.verifiedSecondaryEmailAlreadyExists();
-                }
-
-                return db.deleteEmail(
-                  secondaryEmailRecord.uid,
-                  secondaryEmailRecord.email
-                );
+        await customs.check(request, email, 'accountCreate');
+        await deleteAccountIfUnverified();
```

Comment:
Kind of wish we'd un-wrap & inline all these functions along the way to await-ifying & simplifying things.

But, on the other hand, I guess this kind of serves as a table-of-contents for the functionality. Also makes the conversion safer and smaller.

So, I'm mainly thinking out loud and this is not even a nit.